spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/core/publisher/PubSubPublisherTemplate.java

```diff
@@ -119,7 +121,7 @@ public void onSuccess(String result) {
 				settableFuture.set(result);
 			}
 
-		});
+		}, directExecutor());
```

Comment:
Is `directExecutor()` equivalent to `Runnable::run`?

Maybe you can just use `Runnable::run` here and avoid the guava import.

Reply:
It looks pretty similar, but in this case I just copied it straight from the deprecated implementation: https://github.com/googleapis/api-common-java/blob/master/src/main/java/com/google/api/core/ApiFutures.java#L58

Also note that this isn't Guava - it's `api-common-java`, which I assume comes with all of our client libraries. That being said, I've now added two entries to our checkstyle-exceptions, and I'm not able to find where this `IllegalImport` rule keeps catching these imports.

Reply:
Gotcha, sounds good.